### PR TITLE
More fixes for TPROXY + ipv6

### PR DESCRIPTION
**Pull request by howardjohn**
With this and a few other patches I will send a PR for, I have
IPv6+TPROXY+Distroless+CNI all working together. Currently we don't have
any tests of this but working to expand our environmental testing.

**Please provide a description of this PR:**

### CODE REVIEW

**Comment by howardjohn** [comment]
@lambdai ptal

File: tools/istio-iptables/pkg/cmd/run.go [hunk 1]
```diff
@@ -520,8 +522,10 @@ func (cfg *IptablesConfigurator) run() {
 		cfg.iptables.InsertRule(iptableslog.UndefinedCommand, constants.ISTIOINBOUND, constants.MANGLE, 1,
 			"-p", constants.TCP, "-m", "mark", "--mark", cfg.cfg.InboundTProxyMark, "-j", constants.RETURN)
 		// prevent intercept traffic from envoy/pilot-agent ==> app by 127.0.0.6 --> podip
-		cfg.iptables.InsertRule(iptableslog.UndefinedCommand, constants.ISTIOINBOUND, constants.MANGLE, 2,
+		cfg.iptables.InsertRuleV4(iptableslog.UndefinedCommand, constants.ISTIOINBOUND, constants.MANGLE, 2,
```
Comment (lambdai) [inline tools/istio-iptables/pkg/cmd/run.go:525]: Any reasons not to use AppendVersionedRule ?

File: tools/istio-iptables/pkg/cmd/run.go [hunk 2]
```diff
@@ -520,8 +522,10 @@ func (cfg *IptablesConfigurator) run() {
...
 			"-p", constants.TCP, "-s", "127.0.0.6/32", "-i", "lo", "-j", constants.RETURN)
+		cfg.iptables.InsertRuleV6(iptableslog.UndefinedCommand, constants.ISTIOINBOUND, constants.MANGLE, 2,
+			"-p", constants.TCP, "-s", "::1/128", "-i", "lo", "-j", constants.RETURN)
```
Comment (lambdai) [inline tools/istio-iptables/pkg/cmd/run.go:528]: Why not ::6/128 here? Have we get rid of magic 6-suffix ip in ipv6?

**Review by lambdai — COMMENTED** [review]
LGTM modulo 2 questions

File: tools/istio-iptables/pkg/cmd/run.go (hunk 2 above)
Comment (howardjohn) [reply tools/istio-iptables/pkg/cmd/run.go:528]: thats a bug... good catch

Comment (howardjohn) [reply tools/istio-iptables/pkg/cmd/run.go:528]: we have no tests for this right now but I am working on them btw

File: tools/istio-iptables/pkg/cmd/run.go (hunk 1 above)
Comment (howardjohn) [reply tools/istio-iptables/pkg/cmd/run.go:525]: we don't have InsertVersionedRule.

IMO we should remove all inserts in the future but currently we don't have it

Comment (lambdai) [reply tools/istio-iptables/pkg/cmd/run.go:525]: Got it. +1 on killing the insert!

**Review by lambdai — APPROVED** [review]
LGTM!

**Pull request merged**